app/src/main/java/com/elmeradrianv/shesafe/fragments/MapFragment.java

```diff
@@ -156,11 +171,115 @@ private void showReports(List<Report> reports) {
 
     private void showMarker(String title, double latitude, double longitude, int levelOfRisk) {
         LatLng reportLatLng = new LatLng(latitude, longitude);
-
         MarkerOptions markerOptions = new MarkerOptions()
                 .position(reportLatLng)
                 .title(title).icon(PinAnimation.getNewIconWithLevelOfRisk(levelOfRisk));
         Marker marker = map.addMarker(markerOptions);
         PinAnimation.dropPinEffect(marker);
     }
+
+    @Override
+    public void onMapLongClick(LatLng latLng) {
+        Toast.makeText(getContext(), "Detected", Toast.LENGTH_SHORT).show();
+        showAlertDialogForPoint(latLng);
+    }
+
+    private void showAlertDialogForPoint(final LatLng latLng) {
+        View messageView = LayoutInflater.from(getContext()).
+                inflate(R.layout.new_report_item, null);
+        setupEditDate(messageView);
+        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(getContext());
+        alertDialogBuilder.setView(messageView);
+        HashMap<String, TypeOfCrime> crimes = new HashMap<>();
+        queryTypeOfCrimes(messageView, crimes);
+
+        final AlertDialog alertDialog = alertDialogBuilder.create();
+        alertDialog.setButton(DialogInterface.BUTTON_POSITIVE, "OK",
+                (dialog, which) -> {
+
+                    EditText etDescription = messageView.findViewById(R.id.etDescription);
+                    if (etDescription.getText().toString().isEmpty()) {
+                        Toast.makeText(getContext(), "Please fill the description", Toast.LENGTH_SHORT).show();
+                    } else {
+                        Spinner spinnerCrimes = messageView.findViewById(R.id.sTypeOfCrime);
+                        TypeOfCrime crime = crimes.get(spinnerCrimes.getSelectedItem().toString());
+                        Report report = new Report();
+                        report.put(Report.DESCRIPTION_KEY, etDescription.getText().toString());
+                        String[] pickedDate = ((EditText) messageView.findViewById(R.id.etDate)).getText().toString().split("/");
```

Comment:
clever hack! I think you can just parse the date string though using some of the java utils:

https://stackoverflow.com/a/10028980